### PR TITLE
Recommended Project Structure Guide

### PR DESCRIPTION
**Pull request by odette-elementl**
### Summary & Motivation

New guide that explores the fully featured project and provides recommendations - https://dagster-git-odette-recommendedprojectguide-elementl.vercel.app/guides/dagster/recommended-project-structure

link to old PR: https://github.com/dagster-io/dagster/pull/11968

### CODE REVIEW

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:135]:
```suggestion
This project does not include ops or graphs; if it did, this would be the structure recommendation.
```

🔧 Using commas made this sound like a run-on. Switching to a semi-colon fixes that

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:137]:
```suggestion
We recommend having a `jobs` folder rather than a `jobs.py` file in this situation. Depending on the types of jobs you have, you can create a separate file for each type of job.
```

🔧 Addressing some formatting and grammar issues

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:139]: 🔧 The phrasing here is a bit awkward - what's the recommendation? To define ops and graphs in the same file as the job definition?

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:155]:
```suggestion
So far, we've discussed our recommendations for structuring a large project which contains only one code location. Dagster also allows you to structure a project with multiple definitions. We don't recommend over-abstracting too early; in most cases, one code location should be sufficient. A helpful pattern uses multiple code locations to separate conflicting dependencies, where each definition has its own package requirements (e.g., `setup.py`) and deployment specs (e.g., Dockerfile).
```

🔧 My suggestions here are mostly for clarity and conciseness.

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:161]:
```suggestion
We recommend setting up a separate test folder structure that mirrors the main project (e.g., having a folder for test assets with any applicable subfolders), which contains the unit tests for each of the components of the data pipeline.
```

🔧 Suggesting for punctuation fixes

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:163]:
```suggestion
- Each of the components in Dagster such as assets, sensors, and resources can all be tested separately. Refer to the [Testing in Dagster](https://docs.dagster.io/concepts/testing) documentation for more info.
```

🔧 Oxford comma and completing the second sentence

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended_project_structure.mdx:164]: ❓ This doesn't seem like a complete thought - is there supposed to be more here?

**Review by erinkcochran87 — COMMENTED** [review]
Back to you! This is shaping up nicely.

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (odette-elementl) [reply docs/content/guides/dagster/recommended_project_structure.mdx:74]: Updated if that makes more sense!

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (odette-elementl) [reply docs/content/guides/dagster/recommended_project_structure.mdx:111]: jobs in general, will update

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (yuhan) [inline docs/content/guides/dagster/recommended-project-structure.mdx:140]:
```suggestion
In these situations, we recommend having a `jobs` folder rather than a `jobs.py` file. Depending on the types of jobs you have, you can create a separate file for each types of jobs.
```

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (yuhan) [inline docs/content/guides/dagster/recommended-project-structure.mdx:138]: this doesn't block the pr. but it'd be nice to have a separate PR to actually have a `jobs` folder and include some ops and graphs to demonstrate 1) what we explain here with real code example (as [1] could be a bit abstract to readers), and also 2) when assets vs ops which would follow [this rec](https://docs.dagster.io/guides/dagster/enriching-with-software-defined-assets#when-should-i-use-software-defined-assets).

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (yuhan) [inline docs/content/guides/dagster/recommended-project-structure.mdx:148]: [1]

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (odette-elementl) [reply docs/content/guides/dagster/recommended-project-structure.mdx:138]: sure will put that as a follow up

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (odette-elementl) [reply docs/content/guides/dagster/recommended_project_structure.mdx]: yes

File: docs/content/guides/dagster/recommended_project_structure.mdx
Comment (odette-elementl) [reply docs/content/guides/dagster/recommended_project_structure.mdx]: removing

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended-project-structure.mdx]:
```suggestion
<strong>Got questions about our recommendations, or something to add?</strong> Join our [GitHub discussion](https://github.com/dagster-io/dagster/discussions/8972) to share how you organize your Dagster code.
```

🔧 Adding the link here - I think this is the right discussion (cc @yuhan)

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended-project-structure.mdx:21]:
```suggestion
As your experience with Dagster grows, certain aspects of this guide might no longer apply to your use cases, and you may want to change the structure to adapt to your business needs.
```

🔧 Adding `might` in here - we don't know if it'll apply or not. (Apologies if this was from a suggestion of mine and I missed adding this word in)

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended-project-structure.mdx:11]:
```suggestion
<Note>
  <strong>Got questions about our recommendations or something to add?</strong>{" "}
  Join our <a href="https://github.com/dagster-io/dagster/discussions/8972">GitHub discussion</a> to share how you organize your Dagster code.
</Note>
```

Adding the discussion link, along with formatting this as an HTML link. You'll likely need to run `mdx-format` again for this (bleh)

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended-project-structure.mdx:134]: ❓ Sorry for not catching this - but you link to other folders in the example project except for here. Should this link to `jobs.py`? (https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/jobs.py)

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (erinkcochran87) [inline docs/content/guides/dagster/recommended-project-structure.mdx:166]:
```suggestion
Each of the components in Dagster such as assets, sensors, and resources can all be tested separately. Refer to the [Testing in Dagster](https://docs.dagster.io/concepts/testing) documentation for more info.
```

🔧 Removing an unneeded bullet

**Review by erinkcochran87 — APPROVED** [review]
Please see comments before merging! I found a few things - a missing link, for one - that are all really small. I'm good with this after those things are addressed. Great work! <3

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (yuhan) [reply docs/content/guides/dagster/recommended-project-structure.mdx]: yes it is

File: docs/content/guides/dagster/recommended-project-structure.mdx
Comment (yuhan) [reply docs/content/guides/dagster/recommended-project-structure.mdx:148]: nope it was just the reference for the comment above.

**Review by yuhan — APPROVED** [review]

**Pull request merged**
